From 3a9e9bdd1ceb928392ab4c34e585115b6cfcd279 Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Wed, 11 Oct 2023 10:46:26 +0530
Subject: [PATCH] arm64: perf: increase pmceid array size to 4

Passing ARMV8_PMUV3_MAX_COMMON_EVENTS as nbits argument to
bitmap_from_arr32 iterate upto 0 to 3, but the buf size is 2,
so out of bound error is seen. Hence increasing the size to 4.

Fixes: e75cdd92e98a ("arm64: perf: Expose fine grained L2 cache events")
Change-Id: I959b44256a16124c1fe707c891469fa9e501d9c8
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 arch/arm64/kernel/perf_event.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/kernel/perf_event.c b/arch/arm64/kernel/perf_event.c
index 107a7fd97b3f..f4fe50c5a3cd 100644
--- a/arch/arm64/kernel/perf_event.c
+++ b/arch/arm64/kernel/perf_event.c
@@ -1155,7 +1155,7 @@ static void __armv8pmu_probe_pmu(void *info)
 	struct platform_device *pdev = cpu_pmu->plat_device;
 	u64 dfr0;
 	u64 pmceid_raw[2];
-	u32 pmceid[2];
+	u32 pmceid[4];
 	int pmuver;
 	const char *compatible;
 
-- 
2.34.1

